addes nemy animator controller

oWASDo opened the pull request:
Added animator controller, added laser particle, modify teleport particles

Pierafalcone committed:
Added a script that spawn a test prefab when you select your avatar

AlanCasanova7 committed:
Started some testing on FSM Component Based. Started coding the EnemyShootAI.

AlanCasanova7 committed:
First workig version of BRAIN, A FSM That works with composition and it's easily customizable through the editor.

AlanCasanova7 committed:
Fixed a error that could happen if no AIBehaviours where given to the BRAIN.
Fix: the Init() method of the AIBehaviours was currently called only once when the start of the brain would occour, now it's called when the AIBehaviour is activated for the first time. (COULD CHANGE)
Removed uneccessary dictionary inside the BRAIN class.

Ragdoller committed:
HUD fix + minimization scriptObj (HeadsupDisplay generic-wise)

Pierafalcone committed:
Added cose funzionanti

Pierafalcone committed:
Added cose funzionanti che ora funzionano davvero

Ragdoller committed:
Change : Hud data scriptObjs will be created better, this should fix "player 3 player 3 player 3 " problem. Testing required.

oWASDo committed:
addes nemy animator controller

M-Isonni committed:
fixed errors

aannun committed:
Added reference to main player in server.

aannun committed:
Merge branch 'Develop' of https://github.com/ErrorNullReference/INVASIONnnn into Develop

Ragdoller committed:
Fix Player HUD canvas prefab (HUDMAnager not tracking dataType)

M-Isonni committed:
fixed HUDs

M-Isonni committed:
Merge branch 'HUD' of https://github.com/ErrorNullReference/INVASIONnnn into HUD

M-Isonni committed:
Merge branch 'HUD' into Develop

SamueleAlpino committed:
FSM implemented

AlanCasanova7 committed:
Fixed a typo.

AlanCasanova7 committed:
Merge branch 'FSM' into Develop

AlanCasanova7 commented:
Can't see anything working on main scene. Please fix, animations and particles must be able to be tested on main scene.

AlanCasanova7 committed:
Merge branch 'Develop' into AlanEnemyShooting

AlanCasanova7 committed:
Started adapting AI Scripts to the new FSM System. Spherical Vision and Movement started.

LorenzoBrunori committed:
Added shooting system

LorenzoBrunori committed:
Added shooting system

MatteoNasci committed:
Added SOPRO,VOCASY and GENUtility. Added VOCASY SOs.

Added VOCASY steam implementation scripts and SOs. Removed outdated
scripts. Modified Client to use GENUtility updated version.

MatteoNasci committed:
Added Voice Chat entity prefab and setted references.

MatteoNasci committed:
Moved prefab pos to origin instead or random position

AlanCasanova7 committed:
First Implementation of enemy shooting with composition based FSM. NEED TESTING WITH 2 OR MORE PLAYERS.

AlanCasanova7 committed:
Added option to disable the BRAIN if you are running as a Client.

AlanCasanova7 committed:
HOTFIX: Enemy were able to see through walls.

aannun committed:
Changed custom rigidbody logic.

aannun committed:
Fixed avatar movement treshold.

MatteoNasci committed:
Added specific Steam Transport. Added PacketType for mute messages.

AlanCasanova7 committed:
Merge pull request #7 from ErrorNullReference/AlessioPlayerAnimator

Alessio player animator

AlanCasanova7 committed:
Merge pull request #6 from ErrorNullReference/AIShootingSystem

Ai shooting system

MatteoNasci committed:
Updated dlls to latest version.

Modified steam transport scriptable object to fix a bug where a
nullreferenceexception would be thrown at Editor time when the
scriptable object is loaded

MatteoNasci committed:
Added chat entities pool, modified voicechatspawner and added to VCT scene

MatteoNasci committed:
Added full compatibility of Handler with pools cycle

MatteoNasci committed:
Renamed assets. Updated dlls. Added comments to VoiceChatSpawner.

MatteoNasci committed:
Deleted redundant scene, added voicechatspawner to TVCI scene.

MatteoNasci committed:
Copied Lobby scene into TVCI, ready for practical tests.

MatteoNasci committed:
Hotfix: fixed bug with steam trasnport where data buffer would have length 0.

MatteoNasci committed:
Removed byte array allocation for ismuted msg, now it uses a buffer.

AlanCasanova7 committed:
Merge branch 'Develop' into developVOCASY

AlanCasanova7 committed:
Merge pull request #8 from ErrorNullReference/developVOCASY

Develop vocasy

M-Isonni committed:
Merge branch 'TestAlan' into TestDevelopAlan

# Conflicts:

M-Isonni committed:
fixed Commands

M-Isonni committed:
fixed commands

aannun committed:
Added events relative to user status.

MatteoNasci committed:
Prepare for remote develop pull

M-Isonni committed:
Fixed and optimized

MatteoNasci committed:
conflicts resolve

Pierafalcone committed:
fixed avatars prefab

M-Isonni committed:
Merge branch 'AlanEnemyShooting' into Develop

M-Isonni committed:
Merge branch 'Develop' of https://github.com/ErrorNullReference/INVASION into Develop

MatteoNasci committed:
Updated dlls and implemented new network events in Voice Chat system.

MatteoNasci committed:
Merge branch 'Develop' of https://github.com/ErrorNullReference/INVASION into developVOCASY

MatteoNasci committed:
Updated SOPRO dll to latest version

MatteoNasci committed:
Updated SOPRO

MatteoNasci committed:
Implemented SOPRO SOs and fixed some stuff.

Removed a lot of allocs when creating byte arrays / bytepackets, now
most of them are chached. Switched all BitConverter calls with
ByteManipulator (faster and no alloc). Switched Layermask / animator
string prop names with LayerMaskHolder / AnimatorPropertyHolder.
UnityEvents switched with SO events. Added 2 new pools that can be used
with non unity classes. More stuff that I don't remember.

MatteoNasci committed:
fixed ProjectileTest scene by adding pool

MatteoNasci committed:
Added generic array pool, modified Client to pool all packets.

MatteoNasci committed:
Fixed Enemy prefabs to match how it works now in Develop.

MatteoNasci committed:
Removed some more allocs, now only empty byte arrays are left.

MatteoNasci committed:
Fixed bug with AI prefabs launching null refs.

Implemente sopro events fro client and added player container.

MatteoNasci committed:
Null check to avoid null ref exception.

MatteoNasci committed:
Updated SOPRO dlls and generated code. Fixed and simplified  more stuff.

Fixed expecially AI scripts by removing unity events and by giving those
who need it a Scriptable object containing all active players. Removed a
lot of physics raycast, in their place used Collider.Raycast (version
that checks only that single collider). More stuff that I don't
remember.

MatteoNasci committed:
Finished enemy prefab fix

MatteoNasci committed:
Removed all byte[] allocs (for real now). Updated client.

Client can now manage non size fixed payloads by handling data start
index and data length. Previously the voice data packets were messed up
because of this.

MatteoNasci committed:
Reworked ai system, removed most unityevents and AIUpdate.

Reworked a lot of singletons to become scriptable objects. NetId now is
an int unique to the object rather than a byte value. Improved scripts.
Updated SOPRO. Created new SO container dictionary that holds all active
network GameNetworkObbject.

MatteoNasci committed:
NetIdDispenser default id value changed to int.MinValue

MatteoNasci committed:
Further improvet Client (Transform packets). Adjusted prefabs and scenes.

MatteoNasci committed:
Fixed NetIdDispenser internal id reset when reaching the max value.

MatteoNasci committed:
Updated SOPRO, fixed packets.

Generalized net objects transform packets. Fixed player packets, now
they will manage the optional attached data on top of the transform data

AlanCasanova7 committed:
Merge branch 'Develop' into lobbyUI

AlanCasanova7 committed:
Merge pull request #9 from ErrorNullReference/lobbyUI

lobbyUI merge

AlanCasanova7 committed:
Merge branch 'Develop' into overallFixesDevelop

AlanCasanova7 committed:
Merge pull request #11 from ErrorNullReference/overallFixesDevelop

Overall fixes develop

AlanCasanova7 committed:
Revert "Overall fixes develop"

AlanCasanova7 committed:
Merge pull request #12 from ErrorNullReference/revert-11-overallFixesDevelop

Revert "Overall fixes develop"

AlanCasanova7 committed:
Trying to fix OverallFix Merging issue.

AlanCasanova7 committed:
Random Testing Commit

AlanCasanova7 committed:
Hopefully this will fix it.

MatteoNasci committed:
Fixed broken prefabs, again

MatteoNasci committed:
Updated SOPRO, modified scripts to adapt to modified SOPool.

MatteoNasci committed:
Updated SOPRO (bug fixes), fixed broken SOPRO SOs. Reorganized things.

oWASDo committed:
Add enemy shooting system

oWASDo committed:
Merge branch 'Stirlani' of https://github.com/ErrorNullReference/INVASION into Stirlani

# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.

oWASDo committed:
Fixing enemy shotting system

oWASDo commented:
Adding enemy shooting menager
